src/network/transaction_request/mod.rs

```diff
@@ -20,6 +20,11 @@ pub struct TransactionRequest {
 
 // TODO: Extension trait for `TransactionBuilder`?
 impl TransactionRequest {
+    pub fn with_base(mut self, base: alloy::rpc::types::transaction::TransactionRequest) -> Self {
```

Comment:
Maybe it would be better to implement `From` here? The fact that our type is used as part of composition is an implementation detail, so not sure if we want to make it a part of the public interface.

Reply:
Agreed, `From` makes more sense here due to it being implementation detail.